pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -779,52 +801,151 @@ func (r *RootSyncReconciler) validateRootSecret(ctx context.Context, rootSync *v
 	return validateSecretData(rootSync.Spec.Auth, secret)
 }
 
-func (r *RootSyncReconciler) configureClusterRoleBinding(ctx context.Context, reconcilerRef types.NamespacedName, roleConfig v1beta1.ClusterRoleOverrideSpec) error {
-	// in order to be backwards compatible with behavior before https://github.com/GoogleContainerTools/kpt-config-sync/pull/938
-	// we delete any ClusterRoleBinding that uses the old name.
-	// This ensures smooth migrations for users upgrading past that version boundary.
-	if err := r.deleteClusterRoleBinding(ctx, fmt.Sprintf("%s:%s", core.RootReconcilerPrefix, configsync.RootSyncName)); err != nil && !apierrors.IsNotFound(err) {
-		return errors.Wrap(err, "deleting old binding")
+func (r *RootSyncReconciler) listCurrentRoleRefs(ctx context.Context, rsRef types.NamespacedName) (map[v1beta1.RootSyncRoleRef]client.Object, error) {
```

Comment:
probably complex enough to need a comment explaining what's happening

Reply:
Added a comment